### PR TITLE
Reimplemented GoogleNet

### PR DESCRIPTION
**Pull request by kedardoshi**
I've cleaned up the implementation of GoogleNet. There's still something wrong with the final fully connected layer. Will need someone with a better understanding of dl4j to fix that.

### CODE REVIEW

**Comment by nyghtowl** [comment]
@kedardoshi Thanks for submitting this update. I've done a prelim review and need to download this and confirm no conflicts before merging. I will get that in very soon.

**Comment by kedardoshi** [comment]
@nyghtowl thanks!

File: src/main/java/imagenet/ImageNetGoogleLeNetExample.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    protected static int asynQues = 5;
+    protected static int normalizeValue = 255;
+    protected static double splitTrainTest = 0.8;
+    protected static int seed = 42;
+    protected static Random rng = new Random(seed);
+
+    private static final Logger log = LoggerFactory.getLogger(ImageNetGoogleLeNetExample.class);
+    
+    public static void main(String[] args) throws Exception {
+    	ComputationGraph model = new GoogleLeNet(HEIGHT, WIDTH, CHANNELS, numLabels, SEED, ITERATIONS).init();
```
Comment (nyghtowl) [inline src/main/java/imagenet/ImageNetGoogleLeNetExample.java]: Good consolidation of ImageNetMain and ImageNetStandardExample to run this model. To minimize code duplication in the overall repo, add this line to ImageNetMain under buildModel (line 159) and remove this class.

**Comment by nyghtowl** [comment]
"There's still something wrong with the final fully connected layer." Do you have a gist or more details about what you were seeing that led to this statement?

File: src/main/java/imagenet/ImageNetMain.java [hunk 2]
```diff
@@ -186,7 +186,7 @@ protected void saveAndPrintResults(){
         System.out.println("Total evaluation runtime: " + testTime + " minutes");
         System.out.println("****************************************************");
         if (saveModel) NetSaverLoaderUtils.saveNetworkAndParameters(model, outputPath.toString());
-        if (saveParams) NetSaverLoaderUtils.saveParameters(model, layerNames, paramPaths);
+        if (saveParams) NetSaverLoaderUtils.saveParameters(model, model.getLayerNames().toArray(new String[]{}), paramPaths);
```
Comment (nyghtowl) [inline src/main/java/imagenet/ImageNetMain.java]: Thanks for the fix

File: src/main/java/imagenet/Models/GoogleLeNet.java [hunk 3]
```diff
@@ -38,375 +44,107 @@ public GoogleLeNet(int height, int width, int channels, int outputNum, long seed
...
-                .addLayer("cnn48", new ConvolutionLayer.Builder(new int[]{1, 1})
-                        .nOut(32)
-                        .biasInit(0.2)
-                        .build(), "cnn40", "cnn43", "cnn44", "cnn45")
-                .addLayer("max10", new SubsamplingLayer.Builder(SubsamplingLayer.PoolingType.MAX, new int[]{3, 3}, new int[]{1, 1}, new int[]{1, 1})
-                        .build(), "cnn40", "cnn43", "cnn44", "cnn45")
-
-//                .addLayer("avg2", new SubsamplingLayer.Builder(SubsamplingLayer.PoolingType.AVG, new int[]{3, 3}, new int[]{1, 1}, new int[]{1, 1})
-//                        .build(), "cnn40", "cnn43", "cnn44", "cnn45")
+    private ConvolutionLayer conv1x1(int in, int out, double bias) {
```
Comment (nyghtowl) [inline src/main/java/imagenet/Models/GoogleLeNet.java]: This consolidation was much needed

File: src/main/java/imagenet/ImageNetGoogleLeNetExample.java (hunk 1 above)
Comment (kedardoshi) [inline src/main/java/imagenet/ImageNetGoogleLeNetExample.java]: Not that straightforward :-). Model doesn't abstract out all the common functionality across MultiLayerNetwork and ComputationGraph (like fit(DatasetIterator), setListeners(...), etc.).
org.deeplearning4j.nn.api needs to be cleaned up.

File: src/main/java/imagenet/Models/GoogleLeNet.java (hunk 3 above)
Comment (kedardoshi) [inline src/main/java/imagenet/Models/GoogleLeNet.java]: yup.

**Comment by kedardoshi** [comment]
You should be able to run ImageNetGoogleLeNetExample and see where it fails.

File: src/main/java/imagenet/ImageNetGoogleLeNetExample.java (hunk 1 above)
Comment (nyghtowl) [inline src/main/java/imagenet/ImageNetGoogleLeNetExample.java]: Good point. We still want to consolidate into main.

**Comment by nyghtowl** [comment]
Kedardoshi: The revisions are appreciated and I'm going to merge them in with a note that you are seeing a break still on this model. When there are failures, its important to get from those who see the errors as much detail about them to help save time on tracking the problem down (so many other fires and priorities on then plate). In addition our systems are different and/or we may not have something setup to easily run it and frankly we haven't had access to a platform for that until very recently. So if you have time can you share more details on the error your seeing in the issues section for this repo? It will help expedite a resolution.

**Comment by kedardoshi** [comment]
@nyghtowl will do.

**Pull request merged**